Modules/mmapmodule.c

```diff
@@ -682,6 +682,86 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            PyErr_SetString(PyExc_IOError, "Unknown access mode for mmap object.");
```

Comment:
Not possible to reach here IMHO. And `IOError` is not a right exception type here. I think no need to raise a exception here, a simple comment `// should not get here`, or an assertion `assert(null)`, or `access_str = "UNKNOWN"` is better.

Reply:
Understood. I'll insert an `assert` here.